js/main.js

```diff
@@ -154,61 +157,78 @@ class App extends React.Component {
     return pathname;
   };
 
-  addPaneBatched = (pane) => {
-    if (!this._timeoutID) {
-      this._timeoutID = setTimeout(this.processBatchedPanes, 100);
+  // ------------------ //
+  // batched processing //
+  // ------------------ //
+
+  // store pane to be processed
+  const addPaneBatched = (pane) => {
+    if (!_timeoutID.current) {
+      _timeoutID.current = setTimeout(processBatchedPanes, 100);
     }
-    this._pendingPanes.push(pane);
+    _pendingPanes.current.push(pane);
   };
 
-  processBatchedPanes = () => {
-    let newPanes = Object.assign({}, this.state.panes);
-    let newLayout = this.state.layout.slice();
+  // run processing on queue
+  const processBatchedPanes = () => {
+    // wait until app is mounted
+    if (!mounted.current) {
+      // console.log('not processing');
```

Comment:
can drop log

Reply:
Adapted & amended first commit. (I found it cleaner that way)